src/runtime/internal/ssr.ts

```diff
@@ -69,20 +69,34 @@ export function merge_ssr_styles(style_attribute, style_directive) {
 	return style_object;
 }
 
-export const escaped = {
-	'"': '&quot;',
-	"'": '&#39;',
-	'&': '&amp;',
-	'<': '&lt;',
-	'>': '&gt;'
-};
+const ATTR_REGEX = /[&"]/g;
+const CONTENT_REGEX = /[&<]/g;
+
+/**
+ * Note: this method is performance sensitive and has been optimized
+ * https://github.com/sveltejs/svelte/pull/5701
+ */
+export function escape(html: string, is_attr = false) {
+	if (typeof html !== 'string') return html;
```

Comment:
When is `escape` used with an object?

This line means an object's `.toString()` will end up being used as-is on something like `${escape(obj)}`. This is an attack vector unless we verify all `escape(obj)` calls are on objects of our own.

Consider still escaping the stringified value (which was the previous behavior, by the way):

```suggestion
	if (typeof html !== 'string') html = html.toString();
```

Reply:
updated this and addressed the other comments as well